languages/python/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,49 @@
-# How to implement an Python concept exercise
+# How to implement a Python concept exercise
 
-TODO: describe how to implement a concept exercise for the Python track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise for the Python track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue.
+
+To implement a concept exercise, the following files need to be created:
+
+```
+languages
+└── Python
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .meta
+                |   ├── config.json
+                |   └── <SLUG>.py
+                ├── src
+                |   └── <NAME>.py
+                └── test
```

Comment:
We've been using `<NAME>_test.py` and keeping it immediately beside the <NAME>.py` file ... I'm not sure we need to change that pattern.

Reply:
Agree. I asked it in a comment before, not sure if that comment is still visible. In any case, changed accordingly.

Reply:
Not sure if my comment here is redundant to the one above, but just in case: I'm in favor of avoiding major structural changes unless there's a compelling reason to do so. The Python track has a good deal of scripts/tools that rely on this structure, and changing it arbitrarily creates additional work.